src/index.js

```diff
@@ -0,0 +1,63 @@
+import 'dotenv/config';
+import express from 'express';
+import userRouter from './routes/userRouter';
+import groupRouter from './routes/groupRouter';
+import authRouter from './routes/authRouter';
+import loginRouter from './routes/loginRouter';
+import logger from './util/logger';
+import { internalErr, notFound } from './util/constant';
+import { unless, stringifyBody } from './util/util';
+import cors from 'cors';
+
+const app = express();
+
+const domainsWhitelist = process.env.CORS_WHITELIST.split(',');
+const corsOptions = {
+    origin: (origin, callback) => {
+        if (domainsWhitelist.indexOf(origin) !== -1) {
+            callback(null, true);
+        } else {
+            callback(new Error(`origin: ${origin} isn't allowed by CORS`));
+        }
+    }
+};
+
+app.use(cors(corsOptions));
+
+app.use(express.json());
+
+app.use(unless('/login', authRouter));
```

Comment:
interesting approach, but maybe create separate route for /users and /groups with auth middleware. And attach it to the root router